extension/src/background/ducks/session.ts

```diff
@@ -115,10 +139,18 @@ export const hasPrivateKeySelector = createSelector(
   sessionSelector,
   async (session) => {
     const isHardwareWalletActive = await getIsHardwareWalletActive();
-    return isHardwareWalletActive || !!session.privateKey.length;
+    return isHardwareWalletActive || !!session?.privateKey?.length;
   },
 );
 export const privateKeySelector = createSelector(
   sessionSelector,
-  (session) => session.privateKey,
+  (session) => session.privateKey || "",
+);
+export const passwordSelector = createSelector(
+  sessionSelector,
+  (session) => session.password,
+);
+export const migratedMnemonicPhraseSelector = createSelector(
+  sessionSelector,
+  (session) => session.migratedMnemonicPhrase,
```

Comment:
we need to hold the old mnemonic and the new mnemonic for a little while, so storing this in a different key